root/static/scripts/release/components/ReleaseRelationshipEditor.js

```diff
@@ -1342,7 +1352,18 @@ const reducer = reducerWithErrorHandling<
                 updateRelationshipState(
                   workRelationship,
                   (newWorkRelationship) => {
-                    newWorkRelationship.entity1 = newWork;
+                    const {entity0, entity1} = newWorkRelationship;
+                    if (
+                      entity0.entityType === 'work' &&
+                      entity0.id === oldWork.id
+                    ) {
+                      newWorkRelationship.entity0 = newWork;
+                    } else if (
+                      entity1.entityType === 'work' &&
+                      entity1.id === oldWork.id
+                    ) {
+                      newWorkRelationship.entity1 = newWork;
+                    }
```

Comment:
Oh, another problem here was that this assumed the newly-created work was always `entity1` on the relationship, which is false.

Reply:
I'm not sure I follow here. Above it has `const oldWork = relationship.entity1` but here you're comparing `entity0` and `entity1` with `oldWork`. Why is it safe to assume earlier that `oldWork` is always `entity1`, but not here?

Reply:
`relationship` above is always a recording-work relationship (passed from `getWorkEditsForEntity`). The code here is updating all relationships associated with the old work, not just recording-work ones. (I added some comments saying the same.)

Reply:
But work-work relationships are the only real case where it can also be entity0, since work sorts last for every other entity-type pair of course.